core/src/main/java/org/mule/runtime/core/privileged/processor/IdempotentRedeliveryPolicy.java

```diff
@@ -90,9 +93,19 @@ public void initialise() throws InitialisationException {
 
     idrId = format("%s-%s-%s", muleContext.getConfiguration().getId(), getLocation().getRootContainerName(), "idr");
     lockFactory = muleContext.getLockFactory();
+    if (store != null && privateStore != null) {
+      throw new InitialisationException(createStaticMessage("Ambiguous definition of object store, both reference and private were configured"),
+                                        this);
+    }
     if (store == null) {
-      store = new ProvidedObjectStoreWrapper<>(null, internalObjectStoreSupplier());
+      if (privateStore == null) //If no object store was defined, create one
```

Comment:
When trying to add tests i realized that redeliveryPolicy was not used anywhere. And the message sources(vm,http,jms) that had it allowed as a child element were not working as expected. After some discussion with PLG we realized that fixing this was another issue alltogether (MULE-13330) and required some thought.
I don't know if maybe it would be better to leave this issue for a while and merge it when the other one is finished.